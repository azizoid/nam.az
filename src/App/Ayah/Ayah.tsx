import { useEffect, useState } from 'react';
import { MdFormatQuote } from 'react-icons/md';

const fetchAyah = async () => {
  const response = await fetch('https://quran.az/api/random');
  const json = await response.json();
  return json;
};

export type AyahApiProps = {
  id: string;
  soorah: number;
  ayah: number;
  content: string;
  content_latinized: string;
  translator: number;
};

const Ayah = () => {
  const [randomAyah, setRandomAyah] = useState<AyahApiProps>({
    id: 'defaultID',
    soorah: 40,
    ayah: 60,
    content:
      "Rəbbiniz dedi: 'Mənə dua edin, Mən də sizə cavab verim. Həqiqətən, Mənə ibadət etməyə təkəbbür göstərənlər Cəhənnəmə zəlil olaraq girəcəklər'.",
    content_latinized: '',
    translator: 4,
  });

  useEffect(() => {
    fetchAyah()
      .then(({ out }) => setRandomAyah(out))
      .catch(error => {
        // console.error(error);
      });
  }, []);

  return (
    <blockquote className="md:block flex flex-col space-y-2 w-4/5 md:w-2/3 mx-auto p-4 my-4 italic border-l-2 text-neutral-600 border-green-400 quote">
      <h4 className="flex justify-start">
        <MdFormatQuote className="text-green-400 mr-4" aria-hidden="true" />
        {randomAyah.soorah} : {randomAyah.ayah}
      </h4>
      <p className="mb-2">{randomAyah.content}</p>
      <cite>
        <a
          href={`https://quran.az/${randomAyah.soorah}?t=${randomAyah.translator}&rel=nam.az`}
          target="_blank"
          rel="noopener noreferrer"
        >
          <small>Surəni Tam Oxu</small>
        </a>
      </cite>
    </blockquote>
  );
};

export default Ayah;
