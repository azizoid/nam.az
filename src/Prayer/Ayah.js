import React, { useState, useEffect } from "react";
import Loader from "./Loader";
import { FaQuoteLeft } from "react-icons/fa";

const Ayah = () => {
  const [ayah, setAyah] = useState({
    loader: true,
    content: {
      s: 40,
      a: 60,
      c:
        "Rəbbiniz dedi: 'Mənə dua edin, Mən də sizə cavab verim. Həqiqətən, Mənə ibadət etməyə təkəbbür göstərənlər Cəhənnəmə zəlil olaraq girəcəklər'.",
    },
  });

  useEffect(() => {
    fetch("https://quran.az/api/random/1/Namaz")
      .then((response) => response.json())
      .then((data) => {
        setAyah({ content: data.out[0], loader: false });
      });
  }, []);

  return ayah.loader ? (
    <Loader />
  ) : (
    <blockquote className="ayah">
      <FaQuoteLeft style={{ color: "#66cc66" }} />
      <cite>
        {ayah.content.s} : {ayah.content.a}
      </cite>
      {ayah.content.c}
      <a
        href={
          "https://quran.az/" +
          ayah.content.s +
          "#" +
          ayah.content.a +
          "?rel=namaz"
        }
        target="_blank"
        rel="noopener noreferrer"
        className="link"
      >
        <br />
        <small>Surəni Tam Oxu</small>
      </a>
    </blockquote>
  );
};

export default Ayah;
