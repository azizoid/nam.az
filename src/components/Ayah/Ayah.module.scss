/* Print Ayah under Prayer time */

.ayah {
    position: relative;
    text-align: left;
    padding: 0 0 2em 38px;
    border: none;
    margin: 20px auto 0;
    max-width: 800px;
    width: 100%;
    display: block;

    &:after {
        content: "";
        display: block;
        width: 2px;
        height: 100%;
        position: absolute;
        left: 0;
        color: #66cc66;
        top: 0;
        background: -moz-linear-gradient(
            top,
            #66cc66 0%,
            #66cc66 60%,
            rgba(255, 255, 255, 0) 100%
        );
        background: -webkit-linear-gradient(
            top,
            #66cc66 0%,
            #66cc66 60%,
            rgba(255, 255, 255, 0) 100%
        );
        /* background: linear-gradient(to bottom,#66cc66 0%,#66cc66 60%,rgba(255,255,255,0) 100%); */
    }

    cite {
        display: block;
        font-size: 16px;
        line-height: 1.3em;
        font-weight: 700;
        font-style: normal;
        margin-top: 0.7em;
        letter-spacing: 0;
        font-style: italic;
    }

    .link {
        text-decoration: none;
    }
}
