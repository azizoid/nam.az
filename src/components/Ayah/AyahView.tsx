import { MdFormatQuote } from 'react-icons/md'

import { AyahApiProps } from './Ayah'

type AyahViewProps = { randomAyah: AyahApiProps }

export const AyahView = ({ randomAyah }: AyahViewProps) => (
  <blockquote className="mx-auto my-4 flex w-4/5 flex-col space-y-2 border-l-2 border-green-400 p-4 italic text-neutral-600 md:block md:w-2/3" id="quote">
    <h4 className="flex justify-start">
      <MdFormatQuote className="mr-4 text-green-400" aria-hidden="true" />
      {randomAyah.soorah} : {randomAyah.ayah}
    </h4>
    <p className="mb-2">{randomAyah.content}</p>
    <cite>
      <a
        href={`https://quran.az/${randomAyah.soorah}?t=${randomAyah.translator}&rel=nam.az`}
        target="_blank"
        rel="noopener noreferrer"
      >
        <small>SurÉ™ni Tam Oxu</small>
      </a>
    </cite>
  </blockquote>
)
