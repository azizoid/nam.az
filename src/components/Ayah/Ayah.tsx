import useSWR from 'swr'

import { fetcher } from '@/utilities/fetcher'

import { AyahView } from './AyahView'

export type AyahApiProps = {
  id: string;
  soorah: number;
  ayah: number;
  content: string;
  content_latinized: string;
  translator: number;
};

export const Ayah = () => {
  const apiUrl = 'https://quran.az/api/random'
  const { data } = useSWR(apiUrl, fetcher, {
    revalidateOnMount: true,
    dedupingInterval: 60 * 60 * 1000, // TTL of 1 hour
  })

  const displayAyahByDefault = {
    id: 'defaultID',
    soorah: 40,
    ayah: 60,
    content:
      'Rəbbiniz dedi: \'Mənə dua edin, Mən də sizə cavab verim. Həqiqətən, Mənə ibadət etməyə təkəbbür göstərənlər Cəhənnəmə zəlil olaraq girəcəklər\'.',
    content_latinized: '',
    translator: 4,
  }

  return <AyahView randomAyah={data ? data.out : displayAyahByDefault} />
}
