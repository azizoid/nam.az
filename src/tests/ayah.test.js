import { shallow } from "enzyme";

import Ayah from "../components/ayah.component";

it("expect to render ayah component", () => {
  const wrapper = shallow(<Ayah />);
  expect(wrapper).toMatchSnapshot();
});
