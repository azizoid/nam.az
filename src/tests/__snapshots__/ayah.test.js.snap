// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render ayah component 1`] = `
<blockquote
  className="ayah"
>
  <FaQuoteLeft
    style={
      Object {
        "color": "#66cc66",
      }
    }
  />
  <cite>
    40
     : 
    60
  </cite>
  Rəbbiniz dedi: 'Mənə dua edin, Mən də sizə cavab verim. Həqiqətən, Mənə ibadət etməyə təkəbbür göstərənlər Cəhənnəmə zəlil olaraq girəcəklər'.
  <a
    className="link"
    href="https://quran.az/40#60?rel=namaz"
    rel="noopener noreferrer"
    target="_blank"
  >
    <br />
    <small>
      Surəni Tam Oxu
    </small>
  </a>
</blockquote>
`;
